File: Foundation/lec15/subseq.java
// pre recurssion
class subseq{
    public static void subseq(String ques,String ans) {
        if(ques.length()==0){
            System.out.println(ans);
            return;
        }
        char ch = ques.charAt(0);
        String roq = ques.substring(1);
        subseq(roq,ans+ch);
        subseq(roq,ans);
    }
    public static void main(String[] args) {
        subseq("abc",".");
    }
}

File: Foundation/lec18/subseq.cpp
#include<iostream>
#include<vector>
using namespace std;
vector<string> subseq(string ques){
    if(ques.size()==0){
        vector<string> base;
        base.push_back("");
        return base;
    }
    char ch=ques[0];
    string roq=ques.substr(1);
    vector<string> recAns= subseq(roq);
    vector<string> myAns;
        for(int i=0;i<recAns.size();i++){
            myAns.push_back(ch + recAns[i]);
            }
        for(int i=0;i<recAns.size();i++){
            myAns.push_back(recAns[i]);
        }
    return myAns;
}


int main(){
   vector<string> ans=subseq("abcd");
   for(int i=0;i<ans.size();i++){
       cout<<ans[i]<<endl;
   }
return 0;
}